shopfloor_batch_automatic_creation/actions/picking_batch_auto_create.py

```diff
@@ -104,11 +104,17 @@ def _search_pickings(self, picking_types, user=None, commercial_partner=None):
         # We can't use a limit in the SQL search because the 'priority' fields
         # is sometimes empty (it seems the inverse StockPicking.priority field
         # mess up with default on stock.move), we have to sort in Python.
-        return self.env["stock.picking"].search(
+        pickings = self.env["stock.picking"].search(
             self._search_pickings_domain(
                 picking_types, user=user, commercial_partner=commercial_partner
             )
         )
+        # The selected pickings need to match the type of the first picking found.
+        first_picking_type = fields.first(pickings).picking_type_id
+        filtered_pickings = pickings.filtered(
+            lambda p: p.picking_type_id == first_picking_type
+        )
+        return filtered_pickings
```

Comment:
picking_types is given as an argument to the method so you can add it to the domain in the search method 

another question what will be with other pickings? for me more reasonable is kept this method us is and modify the place where _search_pickings called, and create a batch per picking type

Reply:
The search method already had picking_types as an argument before didn't it?
If you refer to the evaluation of first_picking_type, we want the picking type of the first filtered picking (and not the first picking type available).

Regarding the other pickings, the purpose of this PR is to simply make sure that the batch generated here contains pickings of the same type. The original implementation didn't handle the other pickings either.

Reply:
my idea was to modify returning domain, but the issue is at that moment we don't know which pickings type is found so, ok, the code is LG
but I don't get functional POV,
what is happening with picking which was filtered, they stayed not proceeded? the method called again and another picking type selected? what if we always will have 2 types in the set and another type never will proceed?

Reply:
@jbaudoux thoughts?